### PR TITLE
feat(launch): add create_run_queue to public API

### PR DESCRIPTION
**Pull request by nickpenaranda**
Fixes
-----
- Fixes WB-14240 (partial)

Description
-----------

### <samp>🤖 Generated by Copilot at d53a33a</samp>

This pull request adds the ability to create launch run queues with different resources and access levels using the public API in `wandb/apis/public.py`. It also implements the internal API for managing default resource configurations for run queues in `wandb/sdk/internal/internal_api.py` and exposes it to the public API in `wandb/apis/internal.py`.

Testing
-------
How was this PR tested?

Checklist
-------
- [x] Include reference to internal ticket "Fixes WB-14240" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### <samp>🤖 Generated by Copilot at d53a33a</samp>

> _To launch runs with different resources_
> _We added some methods and courses_
> _The `Api` class_
> _Can now create and pass_
> _Default configs with `config_id` sources_

### CODE REVIEW

File: wandb/apis/public.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
     Any,
     Dict,
     List,
+    Literal,
```
Comment (dmitryduev) [inline wandb/apis/public.py:29]:
```suggestion
```

Comment (dmitryduev) [reply wandb/apis/public.py:29]: beware py37 :( Gotta do

```python
if sys.version_info >= (3, 8):
    from typing import Protocol
else:
    from typing_extensions import Protocol
```

**Review by dmitryduev — APPROVED** [review]
Thanks @nickpenaranda. One comment, then in it goes!

**Comment by nickpenaranda** [comment]
Thanks for the review @dmitryduev ! I have one missing piece of functionality to implement (guard w/ backend capability introspection) -- would you like me to re-request review after I implement it?

**Comment by dmitryduev** [comment]
nah, feel free to merge once it's ready

**Pull request merged**